src/styles/primeng-theme/components/_button.scss

```diff
@@ -1,107 +1,93 @@
-        box-shadow $field-animation-speed ease-in, color $field-animation-speed ease-in;
+    transition: background-color $field-animation-speed ease-in, 
+                box-shadow $field-animation-speed ease-in,
+                color $field-animation-speed ease-in;
 }
 
-.ui-button {
-    @extend #buttons;
-
-    @each $button, $states in $buttons-map {
-        @if ($button == "basic") {
-            @each $label, $props in $states {
-                @include button-properties($label, $props);
-            }
-        } @else {
-            &[#{$button}] {
-                @each $label, $props in $states {
-                    @include button-properties($label, $props);
-                }
-            }
-        }
+#button-raised {
+    @each $label, $props in $button-raised {
+        @include button-properties($label, $props);
     }
 
-    &[disabled] {
-        cursor: inherit;
-    }
-
-    &[iconButton],
-    &[actionButton] {
-        border-radius: 50%;
-        border: none;
-        transition: background-color $field-animation-speed;
+    .ui-button-text {
+        font-weight: 500;
+        text-transform: uppercase;
     }
+}
 
-    &[iconButton] {
-        width: $button-height;
-
-        &[flat]:not([disabled]) {
-            .ui-c {
-                color: $body-font-color;
-            }
+#button-raised-disabled {
+    @include button-state(map-get($button-raised, disabled));
+    @extend #button-disabled;
+}
 
-            &.ui-state-default:not(:hover) {
-                background: none;
-                box-shadow: none;
-            }
-        }
+#button-flat {
+    @each $label, $props in $button-flat {
+        @include button-properties($label, $props);
+    }
+}
 
-        &.inverse {
-            &:disabled {
-                color: rgba($black, 0.2);
-            }
+#button-flat-disabled {
+    @include button-state(map-get($button-flat, disabled));
+    @extend #button-disabled;
+}
 
-            &:not([disabled]) {
-                &.ui-state-default {
-                    background: none;
-                    box-shadow: none;
-                    color: $white;
-                }
+#button-disabled {
+    cursor: inherit;
+}
 
-                &:hover {
-                    background: rgba($black, 0.1);
-                }
+.ui-button {
+    @extend #button-basic;
+    @extend #button-raised;
 
-                [class^="ui-button-icon"],
-                .ui-button-text,
-                .ui-c {
-                    color: $white;
-                }
-            }
-        }
+    &[disabled] {
+        @extend #button-raised-disabled;
     }
 
-    &[actionButton] {
-        font-size: $action-button-icon-size;
-        height: $action-button-height;
-        width: $action-button-height;
-
-        .ui-button-text,
-        .ui-c {
-            color: $action-button-color;
-        }
-
-        &.ui-state-default {
-            background: $blue;
-        }
+    &[flat] {
```

Comment:
@hmoreras I didn't get rid of the flat attribute for buttons for 2 reasons:

1. There are too many flat attr around
2. I think we should create a directive flat that wrap the ripple directive we have.